client/src/search-utils.ts

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect } from "react";
+
+export type SearchProps = {
+  inputValue: string;
+  onChangeInputValue: (value: string) => void;
+  isFocused: boolean;
+  onChangeIsFocused: (isFocused: boolean) => void;
+};
+
+export function useFocusOnSlash(
+  inputRef: React.RefObject<null | HTMLInputElement>
+) {
+  useEffect(() => {
```

Comment:
Would you mind switching to the `React.useEffect` style (and remove the function import)? I know we, unfortunately, employ both styles throughout the code. But I try to make all "new code" I touch always import the namespace. There's no benefit in importing a specific function from `"react"` since it won't ever tree-shake something less. By using `React.useState` and `React.useCallback` etc. we always get a namespace which makes for fewer module-globals and it also means that any diff that needs a new React function creates a diff that doesn't have to also edit line 1.

Reply:
Oh yeah I did notice you starting to use the namespaced version. I actually feel the exact opposite about it, which is probably largely aesthetic but my brain is also generating rationalisations for me. One that I like is: "component code is react code, hence it makes sense to treat it as std-library-esque globals, the rock bottom of things one needs to know when reading the code. Wdyt? To me, and again that might be aesthetics, the `React.` prefixed version reads a lil noisy.

Reply:
I guess I want to be as close to Svelte as possible on the language-library integration continuum, where Svelte just goes for full on "language API becomes component state management API" (which I think might be a tad too far but that's a different topic)

Reply:
wait I think I have a more general reasoning here which convinces me more: I go for non-namepsaced when naming of the imports is unambiguous enough (which ofc is context-specific) and you might use the import multiple times. So I do sometimes namespace unambiguous things, because I don't want to repeat every usage in the imports. But yeah for React hooks I pretty much go 100% without namespace. Again, probably aesthetics as well and I'm also more in favor of consistency over building a bikeshed empire on a hill which I don't want to die on

Reply:
I hear ya. Totally get it. 
In Python, where I'm arguably from, namespacing is hugely more popular. But even there, sometimes, you ditch it when it's something you don't want to type and the name is so obvious that you don't need a namespace. E.g. `from pathlib import Path`. But I think `from datetime.datetime import now` is ugly or `from os.path import join` because what you end up with is not standing out enough. 

This can be said for `useState` and `useEffect`. They're like `Path` (in Python) where a regular code reader doesn't even have to think to mentally parse that. 

The only technical advantage, from "petty" point of view, with namespacing is that your diffs are less noisy and less chance of a merge conflict if you just import namespaces at the top.

Reply:
I think I got some of my motivation from Kent C Dobbs here: https://epicreact.dev/importing-react-through-the-ages/
He lays good arguments for using the full `import * as React from 'react'` 
But where he agrees with [Dan](https://twitter.com/dan_abramov/status/1308739731551858689) and the core React team is that the only one to avoid is `import React from "react"` which I believe we see lots of in our existing code today. Sigh. It's hard to live. 

Let's agree to not agree. Not because we're stubborn, but more that we make this a non-issue. **Let's not make it a review argument at all.** I.e. it's as unimportant as indentation that prettier takes care of. No rewrites necessary and no need to change what already was for a different style just because someone else "told you".
What do you think? Is that a good way of thinking about it?

Having said that, I would r- if someone did:
```js
const {read} = require('content/document');
```
because that's a definite case where the name is too "vague" (unless the file is tiny and doesn't do much else).

Reply:
I guess a difference between me and Kent is that I like to rely on my tools for that, i.e. IntelliJ mostly does the correct thing for me when I add an import like this:
https://user-images.githubusercontent.com/4051932/120617949-e00d4980-c45a-11eb-9e59-980d5aa15fe1.mov

I do encounter cases where it doesn't do the right thing and I probably tend to check the imports after the IDE auto-imported something, so I guess I have internalized that tool based slow-down. I see the point and I appreciate it.

What is also interesting about this issue is that there is the writing vs reading difference. For reading hooks I'd probably always prefer non-namespaced but for writing I can certainly appreciate the import-hassle point. I would optimize for reading though so my motivated reasoning is still motivated here.

With `read` I totally agree that it is too ambiguous and it is hard to imagine a context where it wouldn't be. Overall I also agree with your point that this is not too far from indentation discussions and it would be great to have something like prettier with agreeable defaults.